app/org/sagebionetworks/bridge/services/UserProfileServiceImpl.java

```diff
@@ -58,6 +84,67 @@ public User updateProfile(User user, UserProfile profile) {
         return user;
     }
     
+    @Override
+    public void sendStudyParticipantRoster(Study study) {
+        checkNotNull(study, Validate.CANNOT_BE_NULL, "study");
+
+        executorService.submit(new ParticipantRosterGenerator(study));
+    }
+    
+    private class ParticipantRosterGenerator implements Runnable {
+        private final Study study;
+        
+        public ParticipantRosterGenerator(Study study) {
+            this.study = study;
+        }
+        
+        @Override
+        public void run() {
+            try {
+                List<StudyParticipant> participants = Lists.newArrayList();
+
+                // We must iterate over every account in the application, not just the study directory, in case 
+                // a user signed up with a different study. There is no way to search/query for specific custom 
+                // data, we must iterate over all accounts. Will this exceed 30 seconds eventually? It could.
+                // One solution would be to do this in a worker thread and email the results to the researcher.
```

Comment:
Nitpick: Reword this comment to something like "Our solution is to..." to make it clear that this is, in fact, what we're doing, and not some kind of TODO in disguise.